File: app/src/main/java/xyz/klinker/messenger/view/AutoSummaryListPreference.java
package xyz.klinker.messenger.view;

import android.content.Context;
import android.preference.ListPreference;
import android.util.AttributeSet;

public class AutoSummaryListPreference extends ListPreference {

    public AutoSummaryListPreference(Context context) {
        this(context, null);
    }

    public AutoSummaryListPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onDialogClosed(boolean positiveResult) {
        super.onDialogClosed(positiveResult);
        if (positiveResult) {
            setSummary(getSummary());
        }
    }

    @Override
    public CharSequence getSummary() {
        int pos = findIndexOfValue(getValue());
        return getEntries()[pos];
    }
}

File: app/src/main/java/xyz/klinker/messenger/view/AutoSummaryListPreference.kt
package xyz.klinker.messenger.view

import android.content.Context
import android.preference.ListPreference
import android.util.AttributeSet

class AutoSummaryListPreference : ListPreference {

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
    constructor(context: Context) : super(context)

    override fun onDialogClosed(positiveResult: Boolean) {
        super.onDialogClosed(positiveResult)
        if (positiveResult) {
            summary = summary
        }
    }

    override fun getSummary() = try {
        entries[findIndexOfValue(value)]
    } catch (e: Exception) {
        ""
    }
}